Components/Sensors/MTK3339.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using Scarlet;
+using Scarlet.Components;
+using Scarlet.IO;
+using Scarlet.IO.BeagleBone;
+
+namespace Scarlet.Components.Sensors
+{
+    public class MTK3339 : ISensor
+    {
+        const string UPDATE_200_MSEC = "$PMTK220,200*2C\r\n";
+        const string MEAS_200_MSEC = "$PMTK300,200,0,0,0,0*2F\r\n";
+        const string GPRMC_GPGGA = "$PMTK314,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0*28\r\n";
+        const string QUERY_GPGSA = "$PSRF103,02,01,00,01*27\r\n";
+
+        public float Latitude
+        {
+            get; private set;
+        }
+        public float Longitude
+        {
+            get; private set;
+        }
+
+        IUARTBus UART;
+
+        public MTK3339(IUARTBus uart)
+        {
+            this.UART = uart;
+            WriteString(GPRMC_GPGGA);
+            Thread.Sleep(1);
+            WriteString(MEAS_200_MSEC);
+            Thread.Sleep(1);
+            WriteString(UPDATE_200_MSEC);
+            Thread.Sleep(1);
+            if (uart.BytesAvailable() > 0) { uart.Read(uart.BytesAvailable(), new byte[uart.BytesAvailable()]); }
+        }
+
+        //Returns true if the GPS has a fix and false otherwise. 
+        public bool Test() => HasFix();
+
+        public void EventTriggered(object sender, EventArgs e) => throw new NotImplementedException("MTK3339 doesn't have events");
+
+        public void UpdateState() => GetCoords();
+
+        public bool HasFix()
+        {
+            WriteString(QUERY_GPGSA);
+            string[] result = Read();
+            if (result.Length < 3)
+                return false;
+            return result[2] != "1";
+        }
+
+        private string[] Read()
+        {
+            string GpsResult = "";
+            byte PrevChar = 0;
+            while (PrevChar != '\n')
+            {
+                if (UART.BytesAvailable() < 1) { continue; }
```

Comment:
If there isn't anything available, won't this just loop endlessly without delay?